### PR TITLE
Properly escape dots when writing atomProps

### LINKED ISSUE #5466: CXSmiles isn't properly escaping floating point properties
Example structure

image

The Chemaxon smiles that rdkit generates is:

```
O=C(O)CCCCc1cccc2cncn12 |atomProp:1.apKa.4.2:14.apKa.7.54|
```

Note the second . should be escaped (even though the Chemaxon Docs clearly says that it doesn't for properties)

```
OC(=O)CCCCc1cccc2cncn12 |atomProp:1.apKa.4&#46;2:14.apKa.7|
```

The escape code for '.' is `&#46;`

We do parse it properly though.

```
>>> m = Chem.MolFromSmiles("OC(=O)CCCCc1cccc2cncn12 |atomProp:1.apKa.4&#46;2:14.apKa.7|")
>>> for atom in m.GetAtoms():
...   if atom.HasProp("apKa"): print(atom.GetIdx(), atom.GetProp("apKa"))
... 
1 4.2
14 7
```

### PR DESCRIPTION
**Pull request by bp-kelley**
Fixes #5466

### CODE REVIEW

File: Code/GraphMol/SmilesParse/CXSmilesOps.cpp [hunk 1]
```diff
@@ -1300,6 +1300,19 @@ std::string quote_string(const std::string &txt) {
   return txt;
 }
 
+std::string quote_atomprop_string(const std::string &txt) {
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/CXSmilesOps.cpp:1303]: It appears that quoting is different based on what you are quoting, so making a new function made sense.

File: Code/GraphMol/SmilesParse/CXSmilesOps.cpp [hunk 2]
```diff
@@ -1300,6 +1300,19 @@ std::string quote_string(const std::string &txt) {
...
+  // at a bare minimum, . needs to be escaped
+  std::string res;
+  for(auto c : txt) {
+    if(c == '.') {
+      res += "&#46;";
+    } else {
+      res += c;
+    }
+  }
+  return res;
```
Comment (greglandrum) [inline Code/GraphMol/SmilesParse/CXSmilesOps.cpp:1313]: Shorter and more efficient:
```suggestion
  std::string res = txt;
  for(auto &c : res) {
    if(c == '.') {
      c = "&#46;";
    } 
  }
  return res;
```

**Review by greglandrum — CHANGES_REQUESTED** [review]
suggested cleanup

File: Code/GraphMol/SmilesParse/CXSmilesOps.cpp (hunk 2 above)
Comment (bp-kelley) [reply Code/GraphMol/SmilesParse/CXSmilesOps.cpp:1313]: c is a "char" here so this doesn't work.

Comment (greglandrum) [reply Code/GraphMol/SmilesParse/CXSmilesOps.cpp:1313]: oh, that is irritating.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**